parts/k8s/cloud-init/artifacts/cse_main.sh

```diff
@@ -208,3 +208,4 @@ else
       aptmarkWALinuxAgent unhold &
   fi
 fi
+#EOF
```

Comment:
We aren't using the wait_for_file convenience for provision_source.sh (cse_helper) and it's still using a re-implementation of the old wait_for_file with `if [ -f $script_lib ]`:

https://github.com/Azure/aks-engine/pull/1719/files#diff-38350b45804f5298e3f609c9d4f77cacR9

(we can't use wait_for_file because the function gets defined in that file so we can't call it before having checked it's there).

We need to use a slightly different marker for that file though because `#EOF` is part of the function wait_for_file above so it being present in the file doesn't guarantee that the file has been completely written.

Reply:
Good catch